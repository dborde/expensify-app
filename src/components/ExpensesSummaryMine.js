import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => (
  <div>
    {props.expenseCount < 1 && <p>You have no expenses</p>}
    {props.expenseCount === 1 && <p>Viewing {props.expenseCount} expense totaling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>}
    {props.expenseCount >= 2 && <p>Viewing {props.expenseCount} expenses totaling {numeral(props.expensesTotal / 100).format('$0,0.00')}</p>}
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal: selectExpensesTotal(state.expenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
