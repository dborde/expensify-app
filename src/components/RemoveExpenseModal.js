import React from 'react';
import Modal from 'react-modal';

if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

const OptionModal = (props) => (
  <Modal
    isOpen={props.isModalOpen}
    onRequestClose={props.onToggleModal}
    contentLabel="Remove Expense Confirmation"
    closeTimeoutMS={200}
    className="modal"
  >
    <h3 className="modal__title">Remove Expense</h3>
    <p className="modal__body">Are you sure you want to remove <span>{props.expense.description}</span> expense?</p>
    <button className="button" onClick={props.onToggleModal}>Cancel</button>
    <button className="button button--caution" onClick={props.onRemove}>I&apos;m sure</button>
  </Modal>
);
export default OptionModal;
