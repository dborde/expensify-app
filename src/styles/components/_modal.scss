.ReactModalPortal > div {
  opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
  align-items: center;
  display: flex;
  justify-content: center;
  background-color: lighten($grey, 1%);
  transition: opacity 200ms ease-in-out;

  button {
    margin: 0 1rem;
  }
}

.ReactModalPortal .ReactModal__Overlay--after-open {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
  opacity: 0;
}

.modal {
  background: darken($off-white, 1%);;
  color: $grey;
  max-width: 30rem;
  outline: 1px solid darken($off-white, 15%);
  padding: $l-size;
  text-align: center;
}

.modal__title {
  margin: 0 0 $m-size 0;
}

.modal__body {
  margin: 0 0 $l-size 0;
  font-size: 2rem;
  font-weight: 300;
  word-break: break-word;
  span {
    font-weight: bold;
  }
}
