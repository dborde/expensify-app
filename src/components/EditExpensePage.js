import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';
import RemoveExpenseModal from './RemoveExpenseModal';

export class EditExpensePage extends React.Component {
  state = {
    isModalOpen: false
  };
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onToggleModal = () => {
    this.setState(() => ({ isModalOpen: !this.state.isModalOpen }));
  };
  onRemove = () => {
    this.setState(() => ({ isModalOpen: false }));
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button button--secondary" onClick={this.onToggleModal}>Remove Expense</button>
        </div>
        <RemoveExpenseModal
          expense={this.props.expense}
          isModalOpen={this.state.isModalOpen}
          onToggleModal={this.onToggleModal}
          onRemove={this.onRemove}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
