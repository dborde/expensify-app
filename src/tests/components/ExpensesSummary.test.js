import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={1235} />);
  expect(wrapper).toMatchSnapshot();
});
