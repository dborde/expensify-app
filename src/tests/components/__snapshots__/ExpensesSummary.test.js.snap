// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totaling 
    $12.35
  </h1>
</div>
`;

exports[`should render ExpensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;
